conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,443 @@
+            for dirname in dirnames:
+                relpath = os.path.relpath(os.path.join(root, dirname), d)
+                if is_link(os.path.join(root, dirname)):
+                    links.append(relpath)
+                else:
+                    dirs.append(relpath)
+        return files, dirs, links
+
+    @classmethod
+    def is_broken_link(cls, link):
+        if is_link(link):
+            target = os.readlink(link)
+            return not os.path.exists(target)
+        return False
+
+    @classmethod
+    def get_inodes(cls, d, files):
+        inodes = dict()
+        for filename in files:
+            stat = os.stat(os.path.join(d, filename))
+            if stat.st_ino not in inodes:
+                inodes[stat.st_ino] = []
+            inodes[stat.st_ino].append(filename)
+        return inodes.values()
+
+    def compare_dirs(self, d1, d2):
+        files1, dirs1, links1 = self.get_file_list(d1)
+        files2, dirs2, links2 = self.get_file_list(d2)
+        self.assertEqual(files1, files2)
+        self.assertEqual(dirs1, dirs2)
+        self.assertEqual(links1, links2)
+
+        for link in links1:
+            filename1 = os.path.join(d1, link)
+            filename2 = os.path.join(d2, link)
+
+            self.assertTrue(is_link(filename2))
+            target1 = os.readlink(filename1)
+            target2 = os.readlink(filename2)
+            self.assertEqual(target1, target2)
+
+            self.assertEqual(self.is_broken_link(filename1), self.is_broken_link(filename2))
+
+        for filename in files1:
+            filename1 = os.path.join(d1, filename)
+            filename2 = os.path.join(d2, filename)
+
+            # don't try to run on symlinks - they might be broken!
+            stat1 = os.stat(filename1)
+            stat2 = os.stat(filename2)
+            self.assertEqual(stat1.st_size, stat2.st_size)
+            # 1. verify contents
+            self.assertEqual(md5sum(filename1), md5sum(os.path.join(filename2)))
+            # 2. verify timestamps
+            # FIXME: timestamps don't match???
+            #self.assertEqual(stat1.st_mtime, stat2.st_mtime)
+            # 3. verify permissions
+            self.assertEqual(stat.S_IMODE(stat1.st_mode), stat.S_IMODE(stat2.st_mode))
+
+        # compare I-nodes for hard-links
+        inodes1 = sorted(list(self.get_inodes(d1, files1)))
+        inodes2 = sorted(list(self.get_inodes(d2, files2)))
+        self.assertEqual(inodes1, inodes2)
+
+    @classmethod
+    def gen_n_chars(cls, length):
+        return ''.join([random.choice(string.ascii_letters + string.digits) for _ in range(length)])
+
+    @classmethod
+    def gen_filename(cls, level=0):
+        filename = str(uuid.uuid4())
+        ext = cls.gen_n_chars(3)
+        components = [str(uuid.uuid4()) for _ in range(level)]
+        components.append(filename + "." + ext)
+        return os.path.join(*components)
+
+    def gen_files(self, client, count=1, level=1, filesize=1024):
```

Comment:
I would say that there aren't tests that use nested folders and files (level>1)? This would be very important to test.

Reply:
initially I used them, but they failed on CI due to the too long file names (256 character limits). I think I can shorten file names and enable more levels in tests, would be useful.

Reply:
`test_basic` now uses level=3